deepchem/datasets/__init__.py

```diff
@@ -682,6 +682,40 @@ def _update_mean_and_std(self, df):
       save_to_disk(ys, os.path.join(self.data_dir, row['y_sums']))
       save_to_disk(yss, os.path.join(self.data_dir, row['y_sum_squares']))
 
+  def get_grad_statistics(self):
+    """Computes and returns statistics of this dataset
+
+    This function assumes that the first task of a dataset holds the energy for
+    an input system, and that the remaining tasks holds the gradient for the
+    system.
+
+    TODO(rbharath, joegomes): It is unclear whether this should be a Dataset
```

Comment:
I think the main issue preventing this from being a user-space function is the use of `y_n`. I think this can be avoided by just using `y.shape[0]` directly. Not going to refactor this in the current PR, but have added a note so one of us can clean it up in a follow-on.